### DISCUSSION TITLE
How Khoj different from OpenAI Assistant ?

### DISCUSSION THREAD

**Discussion by lakshmajee** [khoj-ai/khoj#613] [General]
Hi ,
im new here. Just started exploring Khoj.
How Khoj different from OpenAI Assistant ?
Does it need to use OpenAI when it comes to chat based on uploaded PDF files ?

**Comment by debanjum** [comment]
Hi @awkward-minion, thanks for looking into Khoj!

> How Khoj different from OpenAI Assistant ?

Khoj is an open-source, always accessible AI assistant. It allows you to search and chat with your personal documents
- You can interact with it from the Web, Desktop, Obsidian or Emacs app (and [Whatsapp](https://khoj.dev/whatsapp) soon)
- You can use [Khoj cloud](https://app.khoj.dev/login) or [self-host it yourself](https://docs.khoj.dev/get-started/setup)
- You can search, chat with it online (using OpenAI or other commercial models) or offline (using open-source models)

>  Does it need to use OpenAI when it comes to chat based on uploaded PDF files ?

Khoj cloud by default uses OpenAI based models to chat with uploaded PDF files. If you self-host Khoj on your machine, you can use also use offline chat model like Mistral.

**Reply by lakshmajee** [reply]
Thank you @debanjum 👋 .
That explains everything, one query. Does this Khoj tool manipulates uploaded PDF files or just handover to OpenAI ChatGPT or similar models for embedding etc ?

**Reply by debanjum** [reply]
Khoj uses a system called RAG that combines a search AI model (to find relevant document snippets in your synced files) and a chat AI model to give personalized responses to you. 

It uses an open-source search model from HuggingFace (not OpenAI) to generate the embeddings used for search. OpenAI models are usually used by Khoj for only the chat AI. 

Of course, everything is configurable and you can use a different search or chat model if you want
